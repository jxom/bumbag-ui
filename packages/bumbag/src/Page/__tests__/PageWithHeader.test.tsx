import * as React from 'react';
import { PageWithHeader } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<PageWithHeader header={<div>this is a header</div>}>Hello world</PageWithHeader>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <PageWithHeader header={<div>this is a header</div>} color="primary">
        Hello world
      </PageWithHeader>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <PageWithHeader header={<div>this is a header</div>} variant="test">
        hello world
      </PageWithHeader>,
      {
        theme: {
          PageWithHeader: {
            variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
          },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <PageWithHeader
        header={<div>this is a header</div>}
        overrides={{
          PageWithHeader: { styles: { base: { backgroundColor: 'red' } } },
        }}
      >
        hello world
      </PageWithHeader>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<PageWithHeader header={<div>this is a header</div>}>hello world</PageWithHeader>, {
      theme: {
        PageWithHeader: { styles: { base: { backgroundColor: 'red' } } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<PageWithHeader header={<div>this is a header</div>}>hello world</PageWithHeader>, {
      theme: {
        PageWithHeader: { defaultProps: { className: 'test', color: 'primary' } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
