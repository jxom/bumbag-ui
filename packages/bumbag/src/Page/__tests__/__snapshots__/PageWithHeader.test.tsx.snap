// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-0 {
  min-height: 100vh;
  position: relative;
}

.emotion-0 .bb-PageWithSidebarSidebarExpandedWrapper {
  top: 60px;
}

.emotion-0 .bb-PageWithSidebarSidebar {
  height: calc(100vh - 60px);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: var(--bb-palette-primary, #000000)!important;
}

.emotion-6 {
  background-color: var(--bb-palette-background, #000000);
  min-height: 60px;
  height: 60px;
  border-bottom: 1px solid var(--bb-palette-white800, #000000);
  z-index: 999;
}

.emotion-6>* {
  height: 100%;
}

<div
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-PageWithHeader"
>
  <div
    class="emotion-3 emotion-1 emotion-5 emotion-6 bb-PageWithHeaderHeader bb-Box bb-DisclosureContent"
    id="test"
  >
    <div>
      this is a header
    </div>
  </div>
  <div
    class="emotion-1 emotion-5 emotion-9 bb-PageWithHeaderContent bb-Box"
  >
    hello world
  </div>
</div>
`;

exports[`overrides styles.base should render correctly 1`] = `
.emotion-0 {
  min-height: 100vh;
  position: relative;
}

.emotion-0 .bb-PageWithSidebarSidebarExpandedWrapper {
  top: 60px;
}

.emotion-0 .bb-PageWithSidebarSidebar {
  height: calc(100vh - 60px);
}

.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-6 {
  background-color: var(--bb-palette-background, #000000);
  min-height: 60px;
  height: 60px;
  border-bottom: 1px solid var(--bb-palette-white800, #000000);
  z-index: 999;
}

.emotion-6>* {
  height: 100%;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageWithHeader"
>
  <div
    class="emotion-3 emotion-1 emotion-2 emotion-6 bb-PageWithHeaderHeader bb-Box bb-DisclosureContent"
    id="test"
  >
    <div>
      this is a header
    </div>
  </div>
  <div
    class="emotion-1 emotion-2 emotion-9 bb-PageWithHeaderContent bb-Box"
  >
    hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-0 {
  min-height: 100vh;
  position: relative;
}

.emotion-0 .bb-PageWithSidebarSidebarExpandedWrapper {
  top: 60px;
}

.emotion-0 .bb-PageWithSidebarSidebar {
  height: calc(100vh - 60px);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-6 {
  background-color: var(--bb-palette-background, #000000);
  min-height: 60px;
  height: 60px;
  border-bottom: 1px solid var(--bb-palette-white800, #000000);
  z-index: 999;
}

.emotion-6>* {
  height: 100%;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageWithHeader"
>
  <div
    class="emotion-3 emotion-1 emotion-2 emotion-6 bb-PageWithHeaderHeader bb-Box bb-DisclosureContent"
    id="test"
  >
    <div>
      this is a header
    </div>
  </div>
  <div
    class="emotion-1 emotion-2 emotion-9 bb-PageWithHeaderContent bb-Box"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-0 {
  min-height: 100vh;
  position: relative;
}

.emotion-0 .bb-PageWithSidebarSidebarExpandedWrapper {
  top: 60px;
}

.emotion-0 .bb-PageWithSidebarSidebar {
  height: calc(100vh - 60px);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: var(--bb-palette-primary, #000000)!important;
}

.emotion-6 {
  background-color: var(--bb-palette-background, #000000);
  min-height: 60px;
  height: 60px;
  border-bottom: 1px solid var(--bb-palette-white800, #000000);
  z-index: 999;
}

.emotion-6>* {
  height: 100%;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageWithHeader"
>
  <div
    class="emotion-3 emotion-1 emotion-5 emotion-6 bb-PageWithHeaderHeader bb-Box bb-DisclosureContent"
    id="test"
  >
    <div>
      this is a header
    </div>
  </div>
  <div
    class="emotion-1 emotion-5 emotion-9 bb-PageWithHeaderContent bb-Box"
  >
    Hello world
  </div>
</div>
`;

exports[`theming styles.base should render correctly 1`] = `
.emotion-0 {
  min-height: 100vh;
  position: relative;
}

.emotion-0 .bb-PageWithSidebarSidebarExpandedWrapper {
  top: 60px;
}

.emotion-0 .bb-PageWithSidebarSidebar {
  height: calc(100vh - 60px);
}

.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-6 {
  background-color: var(--bb-palette-background, #000000);
  min-height: 60px;
  height: 60px;
  border-bottom: 1px solid var(--bb-palette-white800, #000000);
  z-index: 999;
}

.emotion-6>* {
  height: 100%;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageWithHeader"
>
  <div
    class="emotion-3 emotion-1 emotion-2 emotion-6 bb-PageWithHeaderHeader bb-Box bb-DisclosureContent"
    id="test"
  >
    <div>
      this is a header
    </div>
  </div>
  <div
    class="emotion-1 emotion-2 emotion-9 bb-PageWithHeaderContent bb-Box"
  >
    hello world
  </div>
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
.emotion-0 {
  min-height: 100vh;
  position: relative;
}

.emotion-0 .bb-PageWithSidebarSidebarExpandedWrapper {
  top: 60px;
}

.emotion-0 .bb-PageWithSidebarSidebar {
  height: calc(100vh - 60px);
}

.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-6 {
  background-color: var(--bb-palette-background, #000000);
  min-height: 60px;
  height: 60px;
  border-bottom: 1px solid var(--bb-palette-white800, #000000);
  z-index: 999;
}

.emotion-6>* {
  height: 100%;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageWithHeader"
>
  <div
    class="emotion-3 emotion-1 emotion-2 emotion-6 bb-PageWithHeaderHeader bb-Box bb-DisclosureContent"
    id="test"
  >
    <div>
      this is a header
    </div>
  </div>
  <div
    class="emotion-1 emotion-2 emotion-9 bb-PageWithHeaderContent bb-Box"
  >
    hello world
  </div>
</div>
`;
